### PR TITLE
Fix #1636 New instrument gyroscope added

### LINKED ISSUE #1636: New Instrument : GyroScope
**Actual Behaviour**
Currently the app has accelerometer and compass, though both may seem similar to gyroscope, functionality of gyroscope is quite different. Gyroscope provides absolute rotation of device, where accelerpmeter give changes in orientation , whereas compass uses magnetic sensors. Since almost every smart phone has gyroscope built in, this would be a good feature to include

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by neel1998**
Fixes #1636

**Changes**: 

- I have added a new feature gyroscope which uses devices built in sensor to measure rate of rotation
- I have implemented recording functionality for gyroscope as well
- Layout of the device is kept same as accelerometer
- Implementation is similar to Barometer and Lux meter for consistency  

**Screenshot/s for the changes**: 

Screenshot_20190401-031940
Screenshot_20190401-031953
Screenshot_20190401-032005
Screenshot_20190401-032140

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [ ] I have requested reviews from other members

**APK for testing**: 
[gyroscope.apk.zip](https://github.com/fossasia/pslab-android/files/3027190/gyroscope.apk.zip)

### CODE REVIEW

**Comment by neel1998** [comment]
@mariobehling  @CloudyPadmal  @harsh-2711 @cweitat  Kindly review. There few things yet to do. Kindly suggest what should be written in the guide. Also currently I am using png for logo, will convert them to xml soon

File: app/build.gradle [hunk 1]
```diff
@@ -67,6 +67,7 @@ dependencies {
     implementation "com.squareup.picasso:picasso:$rootProject.picassoVersion"
 
     implementation "com.jakewharton:butterknife:$rootProject.butterKnifeVersion"
+    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
```
Comment (CloudyPadmal) [inline app/build.gradle]: This dependency is already defined. We can remove this line from here.

File: app/src/main/java/io/pslab/interfaces/sensorloggers/GyroscopeRecordables.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package io.pslab.interfaces.sensorloggers;
+
+import io.pslab.models.GyroData;
+import io.realm.RealmResults;
+
+public interface GyroscopeRecordables {
+    GyroData getGyroData(long timeStamp);
+
+    void clearAllGyroRecords();
+
```
Comment (ho-dor) [inline app/src/main/java/io/pslab/interfaces/sensorloggers/GyroscopeRecordables.java]: Do we need these extra spaces ?

**Comment by CloudyPadmal** [comment]
Conflicts to resolve :bell:

**Comment by neel1998** [comment]
@CloudyPadmal  I have resolved the merge conflicts. Kindly find the attached apk for testing. The gyroscope implementation is almost complete. There minor changes with settings fragment and logo, which i suppose can be made on the go. If you find the PR valid kindly merge.

**Pull request merged**